File: 206-reverse-linked-list/reverse-linked-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* reverseList(struct ListNode* head) {
    struct ListNode* curr=head;
    struct ListNode* prev=NULL;
    struct ListNode* nextn=NULL;

    while(curr){
        nextn=curr->next;
        curr->next=prev;
        prev=curr;
        curr=nextn;
    }
    head=prev;
    return head;
}

File: 206-reverse-linked-list/reverse-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        prev, curr, nextn = None, head, None
        while curr:
            nextn = curr.next
            curr.next = prev
            prev = curr
            curr = nextn
        head = prev
        return prev
